### DISCUSSION TITLE
Patch file after pouring bottle

### DISCUSSION THREAD

**Discussion by mahrud** [Homebrew/discussions#4573] [Writing Formulae/Casks] [answered]
### Description of issue

Is there an option or segment of a formula that would be executed after a bottle is poured?

In this case creating a new bottle takes in the order of hours, and besides I can't currently rebuild bottles for M1 machines (the formula is on an independent tap). Since the file that needs to be patched is a static file, I was hoping I could add a section to the formula that applies a patch after the (unpatched) bottle is poured.

I'm happy with a kludge-y, temporary workaround for the time being until I rebuild the bottles.

**Comment by carlocab** [comment, marked as answer]
Yes. Use a `post_install` method.

**Reply by mahrud** [reply]
This may be too much to hope for, but is it possible to do a `patch do` call inside it? I will try it soon, but I'm worried calling GNU patch manually will cause brew style to fail.

**Reply by carlocab** [reply]
Not sure -- seems unlikely. But you can give it a try. Otherwise you can try an `inreplace`, oryou can write a small wrapper script that calls GNU patch to trick `brew style`.
